File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        k = len(s) // 2
        left = sorted(s[:k])
        mid = s[k] if len(s) % 2 != 0 else ""
        return "".join(left) + mid + "".join(left[::-1])

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int n = s.length();
        int k = n / 2;
        string left = s.substr(0, k);
        sort(left.begin(), left.end());
        string mid = (n % 2 != 0)? string(1, s[k]): "";
        string right = left;
        reverse(right.begin(), right.end());
        return left + mid + right;
    }
};